src/v/cloud_storage/types.h

```diff
@@ -55,9 +55,7 @@ enum class upload_result : int32_t {
     failed,
 };
 
-enum class manifest_version : int32_t {
-    v1 = 1,
-};
+static constexpr int32_t partition_manifest_version = 1;
```

Comment:
the idea behind the enum was that eventually you could have more than one version and some conditions in the code with branching on v1, v2, etc

since the shadow indexing is GA now, we should preserve backward compatibility